### PR TITLE
Bmp support

### PR DESCRIPTION
**Pull request by mossblaser**
(Per #44)

* [x] Power control
* [x] LED control
* [x] FPGA register reading
* [x] Temperature reading
* [x] Test sequencing

### CODE REVIEW

File: rig/machine_control/bmp_controller.py [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+        """Read ADC data from the BMP including voltages and temperature.
+
+        Returns
+        -------
+        :py:class:`.ADCInfo`
+        """
+        response = self._send_scp(cabinet, frame, board, SCPCommands.bmp_info,
+                                  arg1=BMPInfoType.adc, expected_args=0)
+        data = struct.unpack("<"  # Little-endian
+                             "HHHHHHHH"  # uint16_t adc[8]
```
Comment (mundya) [inline rig/machine_control/bmp_controller.py]: Not serious, why not `8H`?

File: rig/machine_control/bmp_controller.py [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+        Measured voltage on the 1.2 V rail A.
+    v1_2b : float
+        Measured voltage on the 1.2 V rail B.
+    v1_2c : float
+        Measured voltage on the 1.2 V rail C.
+    v1_8 : float
+        Measured voltage on the 1.8 V rail.
+    v3_3 : float
+        Measured voltage on the 3.3 V rail.
+    vpwr : float
```
Comment (mundya) [inline rig/machine_control/bmp_controller.py]: I've been trying to not be too terse with names*: `supply_voltage`?

\* with debatable success.

File: rig/machine_control/scp_connection.py [hunk 3]
```diff
@@ -64,6 +66,10 @@ def send_scp(self, buffer_size, x, y, p, cmd, arg1=0, arg2=0, arg3=0,
         expected_args : int
             The number of arguments (0-3) that are expected in the returned
             packet.
+        timeout : float
+            Additional timeout in seconds to wait for a reply on top of the
+            default specified upon instantiation. (If None, the default is
+            used).
```
Comment (mundya) [inline rig/machine_control/scp_connection.py]: "If None then the default timeout is used with no addition made"?

Comment (mundya) [inline rig/machine_control/scp_connection.py]: Or why not just default to `0.0`?

Comment (mossblaser) [inline rig/machine_control/scp_connection.py]: To allow me to not change the socket timeout unless it is actually requested which I believe is good for the kernel(?)

**Comment by mundya** [comment]
Cool, LGTM! The ordered tests are nice.

**Pull request merged**